src/main/java/com/teethcare/service/impl/booking/AppointmentServiceImpl.java

```diff
@@ -0,0 +1,159 @@
+import com.teethcare.service.*;
+import com.teethcare.utils.ConvertUtils;
+import com.teethcare.utils.PaginationAndSortFactory;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class AppointmentServiceImpl implements AppointmentService {
+
+    private final BookingService bookingService;
+    private final ServiceRepository serviceRepository;
+    private final AppointmentRepository appointmentRepository;
+    private final AccountService accountService;
+    private final JwtTokenUtil jwtTokenUtil;
+    private final ManagerService managerService;
+    private final ClinicService clinicService;
+    private final DentistService dentistService;
+    private final CSService csService;
+    private final FirebaseMessagingService firebaseMessagingService;
+
+    @Override
+    @Transactional
+    public Appointment createAppointment(String jwtToken, AppointmentRequest appointmentRequest) {
+        if (appointmentRepository.findAppointmentByPreBookingId(appointmentRequest.getPreBookingId()) == null) {
+            Appointment appointment = new Appointment();
+            String username = jwtTokenUtil.getUsernameFromJwt(jwtToken);
+            Account account = accountService.getAccountByUsername(username);
+
+            Timestamp now = new Timestamp(System.currentTimeMillis());
+            appointment.setCreateBookingDate(now);
+            if (appointmentRequest.getAppointmentDate() - now.getTime() < 0) {
+                throw new BadRequestException("Appointment Date invalid");
+            }
+            appointment.setAppointmentDate(ConvertUtils.getTimestamp(appointmentRequest.getAppointmentDate()));
+            if (appointmentRequest.getExpirationAppointmentDate().compareTo(appointmentRequest.getAppointmentDate()) < 0) {
+                throw new BadRequestException("Expiration Appointment Date invalid");
+            }
+            appointment.setExpireAppointmentDate(ConvertUtils.getTimestamp(appointmentRequest.getExpirationAppointmentDate()));
+
+            Booking preBooking = bookingService.findBookingById(appointmentRequest.getPreBookingId());
+            if (preBooking == null) {
+                throw new NotFoundException("Booking ID" + appointmentRequest.getPreBookingId() + " not found!");
+            }
+            appointment.setPreBooking(preBooking);
+            if (appointmentRequest.getServiceId() != null) {
+                ServiceOfClinic service = serviceRepository.findByIdAndStatus(appointmentRequest.getServiceId(), Status.Service.ACTIVE.name());
+                List<ServiceOfClinic> services = new ArrayList<>();
+                services.add(service);
```

Comment:
List<ServiceOfClinic> services = List.of(service);
Dùng cái này khi mà tạo list từ elements đã có, nma list này ko modified đc. Nên xài vẫn okie cho gọn